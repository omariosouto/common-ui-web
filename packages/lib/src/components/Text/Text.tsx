import React, { JSX } from "react";

interface TextProps {
  tag?: string;
  className?: string;
  children: React.ReactNode;
}
export function Text({
  children,
  className,
}: TextProps) {
  const Tag = "p" as keyof JSX.IntrinsicElements;

  return <Tag className={className}>{children}</Tag>;
}