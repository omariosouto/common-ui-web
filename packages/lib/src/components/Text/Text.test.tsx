import { render, screen } from "@testing-library/react";
import { describe, it, expect } from 'vitest'
import { Text } from "./Text";

describe('<Text>', () => {
  it('renders as expected', () => {
    render(<Text className="text-2xl">Sample</Text>)
    expect(screen.getByText('Sample')).toBeInTheDocument();

    expect(screen.getByText('Sample')).toMatchSnapshot();
  });
})