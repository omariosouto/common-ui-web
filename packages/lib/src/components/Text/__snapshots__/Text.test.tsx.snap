// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Text> > renders as expected 1`] = `
<p
  class="text-2xl"
>
  Sample
</p>
`;
